src/rpc/mining.cpp

```diff
@@ -529,7 +529,7 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
     const Consensus::Params& consensusParams = Params().GetConsensus();
 
     // Update nTime
-    UpdateTime(pblock, consensusParams, pindexPrev);
+    UpdateTime(pblock, pindexPrev);
     pblock->nNonce = 0;
```

Comment:
You never update nBits here, whereas it should.

Reply:
@MarcoFalke nBits updating depends on whether CreateNewBlock() is called. That is to say, nBits will be updated when CreateNewBlock() is called. It indicates the miner is mining the same height block at the current time at least if CreateNewBlock() is not called. So the nBits needs not to be update, just use cached blocktemplate. 
The following is the condition of calling CreateNewBlock():
```
if (pindexPrev != chainActive.Tip() ||
        (mempool.GetTransactionsUpdated() != nTransactionsUpdatedLast && GetTime() - nStart > 5) ||
        fLastTemplateSupportsSegwit != fSupportsSegwit)
```

Reply:
Then, why would you need to call `UpdateTime` at all?

Reply:
UpdateTime is responsible for updating nTime field.

Reply:
But it is already called in `CreateNewBlock` just before `GetNextWorkRequired`. So if you can skip the `GetNextWorkRequired` in `UpdateTime` (with the reason that it was called twice through CNB), you can also skip `UpdateTime`, no?

Reply:
If you don't update nBits here, then this patch would change behavior for `getblocktemplate` users on testnet, right?  Previously the calculated difficulty could go down if enough time passes (even using a cached block), whereas after this patch that could no longer happen.

Reply:
@sdaftuar Yeah, that is true. I did not consider it before now. So it works although it is  a bad code smell.